katpoint/test/test_projection.py

```diff
@@ -141,14 +195,62 @@ def test_corner_cases(self):
         assert_angles_almost_equal(ae, [0.0, 0.0], decimal=12)
         ae = np.array(self.plane_to_sphere(0.0,  -np.pi / 2.0, 0.0, -1.0))
         assert_angles_almost_equal(ae, [np.pi, 0.0], decimal=12)
+
+    def test_out_of_range_cases(self):
+        """SIN projection: test out-of-range cases."""
+        # SPHERE TO PLANE
+        # Points outside allowed domain on sphere
+        with OutOfRange.set_treatment('raise'):
+            self.assertRaises(OutOfRangeError,
+                              self.sphere_to_plane, 0.0, np.pi, 0.0, 0.0)
+            self.assertRaises(OutOfRangeError,
+                              self.sphere_to_plane, 0.0, 0.0, np.pi, 0.0)
+            self.assertRaises(OutOfRangeError,
+                              self.sphere_to_plane, 0.0, 0.0, 0.0, np.pi)
+        with OutOfRange.set_treatment('nan'):
+            xy = np.array(self.sphere_to_plane(0.0, np.pi, 0.0, 0.0))
+            np.testing.assert_array_equal(xy, [np.nan, np.nan])
+            xy = np.array(self.sphere_to_plane(0.0, 0.0, np.pi, 0.0))
+            np.testing.assert_array_equal(xy, [np.nan, np.nan])
+            xy = np.array(self.sphere_to_plane(0.0, 0.0, 0.0, np.pi))
+            np.testing.assert_array_equal(xy, [np.nan, np.nan])
+        with OutOfRange.set_treatment('clip'):
+            xy = np.array(self.sphere_to_plane(0.0, np.pi, 0.0, 0.0))
+            np.testing.assert_almost_equal(xy, [0.0, -1.0], decimal=12)
+            xy = np.array(self.sphere_to_plane(0.0, 0.0, np.pi, 0.0))
+            np.testing.assert_almost_equal(xy, [-1.0, 0.0], decimal=12)
+            xy = np.array(self.sphere_to_plane(0.0, 0.0, 0.0, np.pi))
+            np.testing.assert_almost_equal(xy, [0.0, 1.0], decimal=12)
+
+        # PLANE TO SPHERE
```

Comment:
Seems like it should be the start of a new test.

Reply:
Agreed.